kansha/user/usermanager.py

```diff
@@ -7,39 +7,29 @@
 # the file LICENSE.txt, which you should have received as part of
 # this distribution.
 #--
-
-from .models import DataUser
-from nagare.namespaces import xhtml
-from nagare import component, i18n
-from ..toolbox import autocomplete
 import random
 from datetime import datetime, timedelta
 
+from nagare.namespaces import xhtml
+from nagare import component, i18n
 
-def get_user_class(source):
-    """ Return User Class for a given source
-
-    Generic method (from peak). Authentication system implements this
-    method.
-
-    In:
-     - ``source`` -- login source (i.e application, google...)
-    Return:
-     - the user class
-    """
-    raise Exception("User class for source %s not found" % source)
-
-
-def get_app_user(username, data=None):
-    """Return User instance"""
-    if not data:
-        data = UserManager().get_by_username(username)
-    klass = get_user_class(data.source)
-    return klass(username, data=data)
+from ..toolbox import autocomplete
+from .models import DataUser
+from .comp import User
 
 
 class UserManager(object):
 
+    @classmethod
+    def get_app_user(cls, username, data=None):
+        """Return User instance"""
+        if not data:
+            data = cls.get_by_username(username)
+        if data.source != 'application':
+            # we need to set a passwd for nagare auth
+            return User(username, 'passwd', data=data)
```

Comment:
No multiple 'return' in a function (only authorized on arguments checks):

if data.source != 'application':
  user = User(username, 'passwd', data=data)
else:
  user = User(username, data=data)

return user